### PR TITLE
[ZEPPELIN-4185] Upgrade Thrift to 0.12.0

### PR DESCRIPTION
**Pull request by r-kamath**
### What is this PR for?
Upgrade Thrift to 0.12.0

### What type of PR is it?
Improvement

### Todos
--

### What is the Jira issue?
[ZEPPELIN-4185](https://jira.apache.org/jira/browse/ZEPPELIN-4185)

### How should this be tested?
* Should pass CI

### Questions:
* Does the licenses files need update? Yes
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

**Comment by prabhjyotsingh** [comment]
Assuming this is generated using `zeppelin-interpreter/src/main/thrift/genthrift.sh` LGTM.

**Comment by jongyoul** [comment]
Just curiosity. Was there any reason to upgrade the version of thrift?

**Comment by r-kamath** [comment]
There is a vulnerability in 0.9.3. Zeppelin is not directly affected.

**Comment by jongyoul** [comment]
Got it. LGTM

**Comment by r-kamath** [comment]
will merge this if there are no more discussions

File: zeppelin-interpreter/src/test/java/org/apache/zeppelin/dep/DependencyResolverTest.java [hunk 1]
```diff
@@ -89,13 +89,13 @@ public void testLoad() throws Exception {
     // load from added repository
     resolver.addRepo("sonatype",
         "https://oss.sonatype.org/content/repositories/agimatec-releases/", false);
-    resolver.load("com.agimatec:agimatec-validation:0.9.3", testCopyPath);
```
Comment (liuxunorg) [inline zeppelin-interpreter/src/test/java/org/apache/zeppelin/dep/DependencyResolverTest.java]: Need to be modified here?

Comment (liuxunorg) [reply zeppelin-interpreter/src/test/java/org/apache/zeppelin/dep/DependencyResolverTest.java]: TestLoad() execution failed causing CI to fail.

Comment (r-kamath) [reply zeppelin-interpreter/src/test/java/org/apache/zeppelin/dep/DependencyResolverTest.java]: @liuxunorg this is an accidental change. However, agimatec-validation 0.9.3 or 0.12.0 are not available and it is addressed in #3381

**Comment by liuxunorg** [comment]
hi @r-kamath , Now zeppelin can't close the interpreter normally through thrift, I am not familiar with thrift, is there a bug in thrift-0.12.0 version?
https://issues.apache.org/jira/browse/ZEPPELIN-4279
https://issues.apache.org/jira/browse/THRIFT-4858

### zeppelin server error
image

### zeppelin interpreter error
image

**Comment by Reamer** [comment]
If have the same messages also at interpreter start.
```
SLF4J: Actual binding is of type [org.slf4j.impl.Log4jLoggerFactory]
 WARN [2019-08-09 14:59:00,428] ({main} NativeCodeLoader.java[<clinit>]:62) - Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
 WARN [2019-08-09 14:59:01,524] ({main} ZeppelinConfiguration.java[create]:158) - Failed to load configuration, proceeding with a default
 INFO [2019-08-09 14:59:01,604] ({main} ZeppelinConfiguration.java[create]:170) - Server Host: 0.0.0.0
 INFO [2019-08-09 14:59:01,605] ({main} ZeppelinConfiguration.java[create]:172) - Server Port: 8080
 INFO [2019-08-09 14:59:01,605] ({main} ZeppelinConfiguration.java[create]:176) - Context Path: /
 INFO [2019-08-09 14:59:01,606] ({main} ZeppelinConfiguration.java[create]:177) - Zeppelin Version: 0.9.0-SNAPSHOT
 INFO [2019-08-09 14:59:01,615] ({main} RemoteInterpreterServer.java[<init>]:159) - Starting remote interpreter server on port 0, intpEventServerAddress: zeppelin-server.spark.svc.cluster.local:12320
 INFO [2019-08-09 14:59:01,683] ({main} RemoteInterpreterServer.java[<init>]:186) - Launching ThriftServer at 10.131.8.14:12321
ERROR [2019-08-09 14:59:18,833] ({pool-3-thread-1} TThreadPoolServer.java[run]:315) - Thrift error occurred during processing of message.
org.apache.thrift.transport.TTransportException
	at org.apache.thrift.transport.TIOStreamTransport.read(TIOStreamTransport.java:132)
	at org.apache.thrift.transport.TTransport.readAll(TTransport.java:86)
	at org.apache.thrift.protocol.TBinaryProtocol.readAll(TBinaryProtocol.java:425)
	at org.apache.thrift.protocol.TBinaryProtocol.readI32(TBinaryProtocol.java:321)
	at org.apache.thrift.protocol.TBinaryProtocol.readMessageBegin(TBinaryProtocol.java:225)
	at org.apache.thrift.TBaseProcessor.process(TBaseProcessor.java:27)
	at org.apache.thrift.server.TThreadPoolServer$WorkerProcess.run(TThreadPoolServer.java:310)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
 INFO [2019-08-09 14:59:19,100] ({pool-3-thread-2} RemoteInterpreterServer.java[createInterpreter]:379) - Instantiate interpreter org.apache.zeppelin.spark.SparkInterpreter
ERROR [2019-08-09 14:59:19,107] ({pool-3-thread-3} TThreadPoolServer.java[run]:315) - Thrift error occurred during processing of message.
org.apache.thrift.transport.TTransportException
	at org.apache.thrift.transport.TIOStreamTransport.read(TIOStreamTransport.java:132)
	at org.apache.thrift.transport.TTransport.readAll(TTransport.java:86)
	at org.apache.thrift.protocol.TBinaryProtocol.readAll(TBinaryProtocol.java:425)
	at org.apache.thrift.protocol.TBinaryProtocol.readI32(TBinaryProtocol.java:321)
	at org.apache.thrift.protocol.TBinaryProtocol.readMessageBegin(TBinaryProtocol.java:225)
	at org.apache.thrift.TBaseProcessor.process(TBaseProcessor.java:27)
	at org.apache.thrift.server.TThreadPoolServer$WorkerProcess.run(TThreadPoolServer.java:310)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by zjffdu** [comment]
@Reamer Any use experience issue you hit ?

**Comment by Reamer** [comment]
I doesn't recognize any issue,which relates directly to this issue. Sometimes the spark-interpreter timeout doesn't work in kubernetes , maybe it's related to this issue.

**Comment by liuxunorg** [comment]
Often encountered, sometimes the interpreter can not be closed and restarted in the note, will not be related to the problem of thrift, the system can be closed when testing, but often can not restart the interpreter online.

**Comment by efeller** [comment]
Seeing same error here:

`org.apache.thrift.transport.TTransportException
	at org.apache.thrift.transport.TIOStreamTransport.read(TIOStreamTransport.java:132)
	at org.apache.thrift.transport.TTransport.readAll(TTransport.java:86)
	at org.apache.thrift.protocol.TBinaryProtocol.readAll(TBinaryProtocol.java:425)
	at org.apache.thrift.protocol.TBinaryProtocol.readI32(TBinaryProtocol.java:321)
	at org.apache.thrift.protocol.TBinaryProtocol.readMessageBegin(TBinaryProtocol.java:225)
	at org.apache.thrift.TBaseProcessor.process(TBaseProcessor.java:27)
	at org.apache.thrift.server.TThreadPoolServer$WorkerProcess.run(TThreadPoolServer.java:310)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)`

The paragraphs run just fine tho.

**Pull request merged**